instrumentation/internal/internal-class-loader/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/classloader/ClassLoaderInstrumentationModule.java

```diff
@@ -26,10 +29,15 @@ public boolean defaultEnabled(ConfigProperties config) {
   }
 
   @Override
-  public boolean isHelperClass(String className) {
-    // TODO: this can be removed when we drop inlined-advice support
-    // The advices can directly access this class in the AgentClassLoader with invokedynamic Advice
-    return className.equals("io.opentelemetry.javaagent.tooling.Constants");
+  public List<String> getAdditionalHelperClassNames() {
+    return Arrays.asList(
+        "io.opentelemetry.javaagent.instrumentation.internal.classloader.BootstrapPackagesHelper",
+        "io.opentelemetry.javaagent.tooling.Constants");
+  }
+
+  @Override
+  public List<String> injectedClassNames() {
+    return getAdditionalHelperClassNames();
```

Comment:
In this case, do the classes are injected in the target CL or is it only their proxies that happen to have the same name ?

Reply:
We don't want to use proxies here, because then we'd go through the same `invokedynamic` recursions again. Therefore the classes are directly injected.